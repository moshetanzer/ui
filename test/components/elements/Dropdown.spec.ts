import { describe, it, expect } from 'vitest'
import { UDropdown } from '#components'
import type { TypeOf } from 'zod'
import ComponentRender from '../component-render'

describe('Dropdown', () => {
  it.each([
    ['with items', { props: { open: true, items: [{ label: 'label1' }, { label: 'label2' }] } }],
    ['with open', { props: { open: true, items: [{ label: 'label1' }, { label: 'label2' }] } }],
    ['with close', { props: { open: false, items: [{ label: 'label1' }, { label: 'label2' }] } }],
    ['with hover mode', { props: { open: true, items: [{ label: 'label1' }, { label: 'label2' }], mode: 'hover' } }],
    ['with disabled', { props: { disabled: true, items: [{ label: 'label1' }, { label: 'label2' }] } }],
    ['with default slot', { slots: { default: () => 'Default slot' }, props: { open: true, items: [{ label: 'label1' }, { label: 'label2', slot: 'label 2 slot' }] } }]
    // @ts-ignore
  ])('renders %s correctly', async (nameOrHtml: string, options: TypeOf<typeof UDropdown.props>) => {
    const html = await ComponentRender(nameOrHtml, options, UDropdown)
    expect(html).toMatchSnapshot()
  })
})
