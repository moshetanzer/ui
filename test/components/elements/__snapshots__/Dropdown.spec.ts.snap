// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dropdown > renders with close correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="" class="relative inline-flex text-left rtl:text-right">
  <div disabled="false" id="headlessui-menu-button-8" aria-haspopup="menu" aria-expanded="false" data-headlessui-state="" class="inline-flex w-full" role="button"><button> Open </button></div>
  <!--v-if-->
</div>"
`;

exports[`Dropdown > renders with default slot correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div class="relative inline-flex text-left rtl:text-right" data-headlessui-state="open">
  <div disabled="false" id="headlessui-menu-button-14" aria-haspopup="menu" class="inline-flex w-full" role="button" aria-expanded="true" data-headlessui-state="open" aria-controls="headlessui-menu-items-15">Default slot</div>
  <div class="z-20 group w-48" style="position: fixed; left: auto; top: 0px; margin: 0px; right: 0px; bottom: auto; transform: translate(0px, 8px);" data-popper-placement="bottom-end">
    <transition-stub enteractiveclass="transition duration-100 ease-out" enterfromclass="transform scale-95 opacity-0" entertoclass="transform scale-100 opacity-100" leaveactiveclass="transition duration-75 ease-in" leavefromclass="transform scale-100 opacity-100" leavetoclass="transform scale-95 opacity-0" appear="true" persisted="false" css="true">
      <div>
        <!--v-if-->
        <div id="headlessui-menu-items-15" aria-labelledby="headlessui-menu-button-14" role="menu" tabindex="0" data-headlessui-state="open" class="relative focus:outline-none overflow-y-auto scroll-py-1 divide-y divide-gray-200 dark:divide-gray-700 ring-1 ring-gray-200 dark:ring-gray-700 rounded-md shadow-lg bg-white dark:bg-gray-800">
          <div class="p-1" role="none"><button class="group flex items-center gap-1.5 w-full px-1.5 py-1.5 text-sm rounded-md text-gray-700 dark:text-gray-200" id="headlessui-menu-item-16" role="menuitem" data-headlessui-state="">
              <!--v-if--><span class="truncate"></span>
              <!--v-if-->
            </button></div>
          <div class="p-1"><button class="group flex items-center gap-1.5 w-full px-1.5 py-1.5 text-sm rounded-md text-gray-700 dark:text-gray-200" id="headlessui-menu-item-17" role="menuitem" data-headlessui-state="">
              <!--v-if--><span class="truncate"></span>
              <!--v-if-->
            </button><button class="group flex items-center gap-1.5 w-full px-1.5 py-1.5 text-sm rounded-md text-gray-700 dark:text-gray-200" id="headlessui-menu-item-18" role="menuitem" data-headlessui-state="">
              <!--v-if--><span class="truncate"></span>
              <!--v-if-->
            </button></div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;

exports[`Dropdown > renders with disabled correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div data-headlessui-state="" class="relative inline-flex text-left rtl:text-right">
  <div disabled="true" id="headlessui-menu-button-13" aria-haspopup="menu" aria-expanded="false" data-headlessui-state="" class="inline-flex w-full" role="button"><button disabled=""> Open </button></div>
  <!--v-if-->
</div>"
`;

exports[`Dropdown > renders with hover mode correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div class="relative inline-flex text-left rtl:text-right" data-headlessui-state="open">
  <div disabled="false" id="headlessui-menu-button-9" aria-haspopup="menu" class="inline-flex w-full" role="button" aria-expanded="true" data-headlessui-state="open" aria-controls="headlessui-menu-items-10"><button> Open </button></div>
  <div class="z-20 group w-48" style="padding-top: 8px; padding-bottom: 8px; position: fixed; left: auto; top: 0px; margin: 0px; right: 0px; bottom: auto; transform: translate(0px, 0px);" data-popper-placement="bottom-end">
    <transition-stub enteractiveclass="transition duration-100 ease-out" enterfromclass="transform scale-95 opacity-0" entertoclass="transform scale-100 opacity-100" leaveactiveclass="transition duration-75 ease-in" leavefromclass="transform scale-100 opacity-100" leavetoclass="transform scale-95 opacity-0" appear="true" persisted="false" css="true">
      <div>
        <!--v-if-->
        <div id="headlessui-menu-items-10" aria-labelledby="headlessui-menu-button-9" role="menu" tabindex="0" data-headlessui-state="open" class="relative focus:outline-none overflow-y-auto scroll-py-1 divide-y divide-gray-200 dark:divide-gray-700 ring-1 ring-gray-200 dark:ring-gray-700 rounded-md shadow-lg bg-white dark:bg-gray-800">
          <div class="p-1" role="none"><button class="group flex items-center gap-1.5 w-full px-1.5 py-1.5 text-sm rounded-md text-gray-700 dark:text-gray-200" id="headlessui-menu-item-11" role="menuitem" data-headlessui-state="">
              <!--v-if--><span class="truncate"></span>
              <!--v-if-->
            </button></div>
          <div class="p-1"><button class="group flex items-center gap-1.5 w-full px-1.5 py-1.5 text-sm rounded-md text-gray-700 dark:text-gray-200" id="headlessui-menu-item-12" role="menuitem" data-headlessui-state="">
              <!--v-if--><span class="truncate"></span>
              <!--v-if-->
            </button></div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;

exports[`Dropdown > renders with items correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div class="relative inline-flex text-left rtl:text-right" data-headlessui-state="open">
  <div disabled="false" id="headlessui-menu-button-0" aria-haspopup="menu" class="inline-flex w-full" role="button" aria-expanded="true" data-headlessui-state="open" aria-controls="headlessui-menu-items-1"><button> Open </button></div>
  <div class="z-20 group w-48" style="position: fixed; left: auto; top: 0px; margin: 0px; right: 0px; bottom: auto; transform: translate(0px, 8px);" data-popper-placement="bottom-end">
    <transition-stub enteractiveclass="transition duration-100 ease-out" enterfromclass="transform scale-95 opacity-0" entertoclass="transform scale-100 opacity-100" leaveactiveclass="transition duration-75 ease-in" leavefromclass="transform scale-100 opacity-100" leavetoclass="transform scale-95 opacity-0" appear="true" persisted="false" css="true">
      <div>
        <!--v-if-->
        <div id="headlessui-menu-items-1" aria-labelledby="headlessui-menu-button-0" role="menu" tabindex="0" data-headlessui-state="open" class="relative focus:outline-none overflow-y-auto scroll-py-1 divide-y divide-gray-200 dark:divide-gray-700 ring-1 ring-gray-200 dark:ring-gray-700 rounded-md shadow-lg bg-white dark:bg-gray-800">
          <div class="p-1" role="none"><button class="group flex items-center gap-1.5 w-full px-1.5 py-1.5 text-sm rounded-md text-gray-700 dark:text-gray-200" id="headlessui-menu-item-2" role="menuitem" data-headlessui-state="">
              <!--v-if--><span class="truncate"></span>
              <!--v-if-->
            </button></div>
          <div class="p-1"><button class="group flex items-center gap-1.5 w-full px-1.5 py-1.5 text-sm rounded-md text-gray-700 dark:text-gray-200" id="headlessui-menu-item-3" role="menuitem" data-headlessui-state="">
              <!--v-if--><span class="truncate"></span>
              <!--v-if-->
            </button></div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;

exports[`Dropdown > renders with open correctly 1`] = `
"<!-- eslint-disable-next-line vue/no-template-shadow -->
<div class="relative inline-flex text-left rtl:text-right" data-headlessui-state="open">
  <div disabled="false" id="headlessui-menu-button-4" aria-haspopup="menu" class="inline-flex w-full" role="button" aria-expanded="true" data-headlessui-state="open" aria-controls="headlessui-menu-items-5"><button> Open </button></div>
  <div class="z-20 group w-48" style="position: fixed; left: auto; top: 0px; margin: 0px; right: 0px; bottom: auto; transform: translate(0px, 8px);" data-popper-placement="bottom-end">
    <transition-stub enteractiveclass="transition duration-100 ease-out" enterfromclass="transform scale-95 opacity-0" entertoclass="transform scale-100 opacity-100" leaveactiveclass="transition duration-75 ease-in" leavefromclass="transform scale-100 opacity-100" leavetoclass="transform scale-95 opacity-0" appear="true" persisted="false" css="true">
      <div>
        <!--v-if-->
        <div id="headlessui-menu-items-5" aria-labelledby="headlessui-menu-button-4" role="menu" tabindex="0" data-headlessui-state="open" class="relative focus:outline-none overflow-y-auto scroll-py-1 divide-y divide-gray-200 dark:divide-gray-700 ring-1 ring-gray-200 dark:ring-gray-700 rounded-md shadow-lg bg-white dark:bg-gray-800">
          <div class="p-1" role="none"><button class="group flex items-center gap-1.5 w-full px-1.5 py-1.5 text-sm rounded-md text-gray-700 dark:text-gray-200" id="headlessui-menu-item-6" role="menuitem" data-headlessui-state="">
              <!--v-if--><span class="truncate"></span>
              <!--v-if-->
            </button></div>
          <div class="p-1"><button class="group flex items-center gap-1.5 w-full px-1.5 py-1.5 text-sm rounded-md text-gray-700 dark:text-gray-200" id="headlessui-menu-item-7" role="menuitem" data-headlessui-state="">
              <!--v-if--><span class="truncate"></span>
              <!--v-if-->
            </button></div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>"
`;
